### DISCUSSION TITLE
Tauri 2 Beta bundle for Linux with "all needed depencies" included (not need to install WebKit2GTK-4.1 etc on all target machines)

### DISCUSSION THREAD

**Discussion by cagecurrent** [tauri-apps/tauri#10026] [Q&A] [answered]
Hi,
I really like Tauri (tried both v1.6 and v2beta, want to go with v2 directly), and I really want to be able to deploy applications built with on on different host, and to start with Ubuntu on x86 being the most important. The problem though is that I will most likely not be able to force an installation of WebKit2GTK-4.1 and other updated packages for Tauri 2 on every target computer.

Building is not a problem, I have gotten that to work in a Docker container on my Macbook (Intel) as well as in my Gitlab pipeline, but when I then try to run the binary on Linux machine it won't work due to the missing libraries. I have tried to export misc .so-files from the Docker container / Gitlab pipeline and bundle them with the binary when I try to install the app at the target Linux machine, but I have not been successful with this.

Anyone that has done this? Is it even possible?

Thanks in advance! /Per

**Comment by FabianLars** [comment]
Did you try the .AppImage bundle? This is pretty much their main purpose.

The appimage will include all dependencies it needs, hence its size, so it works on non-ubuntu distros too, as long as they glibc version is the same or newer, which means you'll have to build on the oldest system you want the app to run on (in ubuntu terms that's at least 22.04 for v2).

I tried many times to make the appimage backwards compatible (for example build on 22.04, run on 20.04) but it never worked out :/

**Reply by cagecurrent** [reply]
Thanks, I didn't know about the .AppImage concept. (y)

Unfortunately the AppImage doesn't work for me, I get the errors I posted below. I guess it's the **glibc** problem you mention.

mobile: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by mobile)
mobile: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by mobile)
mobile: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by mobile)
mobile: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /tmp/.mount_mobilegboCoW/usr/lib/libwebkit2gtk-4.1.so.0)
mobile: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_mobilegboCoW/usr/lib/libwebkit2gtk-4.1.so.0)
mobile: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_mobilegboCoW/usr/lib/libwebkit2gtk-4.1.so.0)
mobile: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_mobilegboCoW/usr/lib/libwebkit2gtk-4.1.so.0)
mobile: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /tmp/.mount_mobilegboCoW/usr/lib/libwebkit2gtk-4.1.so.0)
mobile: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_mobilegboCoW/usr/lib/libwebkit2gtk-4.1.so.0)

**Reply by FabianLars** [reply, marked as answer]
Yeah, that's the glibc problem. There's no easy way around that sadly (if building on older systems doesn't work). Really one of the biggest problems we have with linux.

**Comment by ofer-shaham** [comment]
try this solution:
https://github.com/shunceyb/tauri-glibc-boilerplate/tree/main
